chain-impl-mockchain/src/testing/scenario/fragment_factory.rs

```diff
@@ -183,18 +183,30 @@ impl FragmentFactory {
         &self,
         valid_until: BlockDate,
         owner: &Wallet,
+        signer: &Wallet,
         update_proposal: UpdateProposal,
     ) -> Fragment {
-        self.transaction_with_cert(valid_until, Some(owner), &update_proposal.into())
+        TestTxCertBuilder::new(self.block0_hash, self.fee).make_transaction_different_signers(
+            valid_until,
+            owner,
+            vec![signer],
+            &update_proposal.into(),
+        )
     }
 
     pub fn update_vote(
         &self,
         valid_until: BlockDate,
         owner: &Wallet,
+        signer: &Wallet,
```

Comment:
Why does this needs to separate who will be an `owner` and who will sign this certificate ?
It should be the same person I do not think that we need to separate this entities.

Reply:
Well not necessarily, for catalyst funds we are using bft nodes which have secrets without any funds. We have one account with some funds which will do that kind of transactions. So we need to send transaction using our account with auth for bft leaders. So there is at least one use case for different auth. Maybe the name is misleading....